tests/containers/testenv_prepare.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2013-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman
+# Summary: Running Podman nginx container under systemd
+# Create my_nginx.container file
+# Inform systemd about the new unit file and this creates a nginx.service
+# Start the created nginx service.
+# Verify the status of the nginx service.
+# Publish the port the container is running on.
+# Fetch a predefined website.
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::utils;
+use containers::container_images;
+
+my $systemd_nginx_unit = <<_EOF_;
+[Unit]
+Description=nginx container quadlet file
+
+[Container]
+Image=registry.opensuse.org/opensuse/nginx:latest
+ContainerName=nginx-Quadlet
+PublishPort=80:80
+Volume=/home/nginx/nginx.conf:/etc/nginx/nginx.conf:rw,Z
+Volume=/home/nginx/index.html:/usr/share/nginx/html/index.html:rw,Z
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+sub run {
+    my ($self) = @_;
+    my $unit_name = 'my_nginx';
+
+    select_serial_terminal;
+    my $quadlet = script_output "rpm -ql podman | grep podman/quadlet";
+
+    assert_script_run(' curl --create-dirs -vo /home/nginx/nginx.conf ' . data_url('containers/nginx/') . 'nginx.conf');
+    assert_script_run(' curl --create-dirs -vo /home/nginx/index.html ' . data_url('containers/nginx/') . 'index.html');
+
+    my $podman = $self->containers_factory('podman');
+    $self->{podman} = $podman;
+
+    # create files for generator
+    assert_script_run("podman --version");
+    assert_script_run("$quadlet -version");
+    assert_script_run("echo '$_' >> /etc/containers/systemd/$unit_name.container") foreach (split /\n/, $systemd_nginx_unit);
+    record_info('Unit', script_output("$quadlet -v -dryrun"));
+
+    # start the generator and check whether the files are generated
+    assert_script_run("systemctl daemon-reload");
+
+    # start the container
+    assert_script_run("systemctl start $unit_name.service", timeout => 120);
```

Comment:
```suggestion
    assert_script_run("systemctl enable --now $unit_name.service", timeout => 120);
```
This needs to survive a reboot, so we likely need to enable the service. Starting would not start it at boot time.

Reply:
I encountered an error when attempting to enable it: 'Failed to enable unit: Unit /run/systemd/generator/my_nginx.service is transient or generated.'

From what I understand, if the *.container file includes an [Install] section, the correct links will be generated to automatically start the service.

Reply:
But do the container then still start at system startup? That's a necessary requirement.

Reply:
yes,  the container starts up automatically.